src/Components/Missions.js

```diff
@@ -0,0 +1,65 @@
+import React, { useEffect, useState } from 'react';
+
+import { useDispatch, useSelector } from 'react-redux';
+import { fetchMissionsData, reserveMission, leaveMission } from '../redux/missions/missionsSlice';
+
+const Missions = () => {
+  const dispatch = useDispatch();
+  const missions = useSelector((state) => state.missions.missions);
+  const [currentMissionId, setCurrentMissionId] = useState(null);
+
+  useEffect(() => {
+    dispatch(fetchMissionsData());
+  }, [dispatch]);
+
+  const handleJoinMission = (missionId) => {
+    dispatch(reserveMission(missionId));
+    setCurrentMissionId(missionId);
+  };
+
+  const handleLeaveMission = (missionId) => {
+    dispatch(leaveMission(missionId));
+    setCurrentMissionId(null);
+  };
```

Comment:
- [ ] Kindly ensure the `JoinMission` and `LeaveMission` work perfectly. Make sure the user can be able to join or leave as many missions as he wants. Now, the user can only be able to join one mission at a time, which is not correct according to the requirement. The user must be able to join or leave many missions at once. Kindly fix that.

  - ### YOUR APP BEHAVIOUR
  ![chrome-capture-2023-4-25 (5)](https://github.com/tomdan-ai/Travalers-space-hub/assets/87197412/4525edb8-db62-4963-b0e5-b26e6cbdc4dc)